File: problem92.c
#include<stdio.h>

int next(int n) {
    int total = 0;
    for( ; n > 0; n /= 10) {
        int d = n % 10;
        total += d*d;
    }
    return total;
}

int main() {
    int k, n, count;
    count = 0;
    for (n = 1; n < 10000000; n++) {
        k = n;
        while (k != 1 && k != 89)
            k = next(k);
        if (k == 89) count++;
    }
    printf("%d", count);
    return -1;
}

File: problem92.py
def next(n):
    return sum([int(i)*int(i) for i in str(n)])

TEN_MIL = 10000000

def solve():
    count = 0
    for n in range(1, TEN_MIL):
        if n%100000 == 0:
            print(n)
        while n != 1 and n != 89:
            n = next(n)
        if n == 89:
            count += 1
    return count